packages/fungible-token/bridge-fungible-token/tests/harness.rs

```diff
@@ -1326,4 +1462,130 @@ mod revert {
         assert_eq!(balance, 0);
     }
+
+    #[tokio::test]
+    async fn deposit_with_wrong_token_twice_registers_two_refunds() {
+        let mut wallet = env::setup_wallet();
+        let configurables: Option<BridgeFungibleTokenContractConfigurables> = None;
+        let wrong_token_value: &str =
+            "0x1111110000000000000000000000000000000000000000000000000000111111";
+
+        let config = env::generate_test_config((BRIDGED_TOKEN_DECIMALS, PROXY_TOKEN_DECIMALS));
+
+        let (message, coin, deposit_contract) = env::construct_msg_data(
+            wrong_token_value,
+            FROM,
+            *Address::from_str(TO).unwrap(),
+            config.min_amount,
+            configurables.clone(),
+            false,
+            None,
+        )
+        .await;
+
+        let one = Unsigned256::from(1);
+
+        let (message2,_,_) = env::construct_msg_data(
+            wrong_token_value,
+            FROM,
+            *Address::from_str(TO).unwrap(),
+            config.min_amount + one,
+            configurables.clone(),
+            false,
+            None,
+        )
+        .await;
+
+        // Set up the environment
+        let (
+            test_contract,
+            contract_inputs,
+            coin_inputs,
+            message_inputs,
+            test_contract_id,
+            provider,
+        ) = env::setup_environment(
+            &mut wallet,
+            vec![coin],
+            vec![message, message2],
+            deposit_contract,
+            None,
+            configurables,
+        )
+        .await;
+
+        // Relay the test message to the test contract
+        let receipts = env::relay_message_to_contract(
+            &wallet,
+            message_inputs[0].clone(),
+            contract_inputs.clone(),
+            &coin_inputs[..],
+            &env::generate_variable_output(),
+        )
+        .await;
+
+        // Relay the test message to the test contract
+        let receipts_second = env::relay_message_to_contract(
+            &wallet,
+            message_inputs[1].clone(),
+            contract_inputs.clone(),
+            &coin_inputs[..],
+            &env::generate_variable_output(),
+        )
+        .await;
+
+        let log_decoder = test_contract.log_decoder();
+        let refund_registered_event = log_decoder
+            .decode_logs_with_type::<RefundRegisteredEvent>(&receipts)
+            .unwrap();
+
+        // Verify the message value was received by the test contract
+        let test_contract_balance = provider
+            .get_contract_asset_balance(test_contract.contract_id(), AssetId::default())
+            .await
+            .unwrap();
+
+        let balance = wallet
+            .get_asset_balance(&AssetId::new(*test_contract_id.hash()))
+            .await
+            .unwrap();
+
+        // Verify the message value was received by the test contract
+        assert_eq!(test_contract_balance, 200);
```

Comment:
I may be misunderstanding. Shouldn't this be `(2 * min_amount) + 1` i.e. 201?
I'm going off the other test being 100

Reply:
I think it is the native balance. The messages being sent to the L2 contract contain data about the token that is being minted (or burnt) and it is this data which gets relayed to the L1 contract. But these L2 messages can also contain (separated from the data being relayed) a value of the native asset. In this case the each of the test messages carry 100 of the native L2 asset. 

I am unsure about the original intent of this - my best guess right now is that it is being used as some form of check that only the built messages (not more, not less) are being sent to the L2 contract and no other message is changing the state in unexpected ways.